File: src/main/java/com/example/chatService/entity/Friend.java
package com.example.chatService.entity;

import com.example.chatService.dto.FriendStatus;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "friends")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Friend {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "friend_id", referencedColumnName = "id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private User friend;

    @ManyToOne
    private User blockedBy; // 누가 차단했는지

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private FriendStatus status;
}

File: src/main/java/com/example/chatService/entity/Friend.kt
package com.example.chatService.entity

import com.example.chatService.dto.FriendStatus
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*

@Entity
@Table(name = "friends")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
class Friend(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
        @JsonIgnoreProperties("hibernateLazyInitializer", "handler")
        var user: User? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "friend_id", referencedColumnName = "id", nullable = false)
        @JsonIgnoreProperties("hibernateLazyInitializer", "handler")
        var friend: User? = null,

        @ManyToOne
        var blockedBy: User? = null, // 누가 차단했는지

        @Enumerated(EnumType.STRING)
        @Column(nullable = false)
        var status: FriendStatus? = null
)